polygeist/include/mlir/Dialect/Polygeist/Utils/TransformUtils.h

```diff
@@ -53,19 +53,47 @@ class IfThenElseBodyInterface {
 // Loop Versioning Utilities
 //===----------------------------------------------------------------------===//
 
+/// Represents a loop versioning condition.
+class LoopVersionCondition {
+public:
+  using SCFCondition = Value;
+
+  struct AffineCondition {
+    IntegerSet ifCondSet;
+    SmallVectorImpl<Value> &setOperands;
+  };
+
+  /// Create a loop versioning condition for SCF loops.
+  LoopVersionCondition(SCFCondition scfCond)
+      : versionCondition({&scfCond, nullptr}) {}
```

Comment:
would scfCond be out of scope after the constructor?

Reply:
This has changed to use an Optional